app/services/blacklight/search_service.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+# SearchService returns search results from the repository
+module Blacklight
+  class SearchService
+    include Blacklight::RequestBuilders
+
+    def initialize(blacklight_config)
+      @blacklight_config = blacklight_config
+    end
+
+    # TODO: Can this be private?
```

Comment:
This requires a change to the signature of `SearchBuilder`, so I'd prefer to make that a separate PR. Presently the `SearchBuilder` is delegating `blacklight_config` to it's passed in context which is an instance of `SearchService` at this time.